File: SDKs/twilio.com/twilio_preview/1.36.0/go/pkg/models/operations/fetchmarketplaceavailableaddonextension.go
package operations

import (
"openapi/pkg/models/shared")
var FetchMarketplaceAvailableAddOnExtensionServers = []string{
	"https://preview.twilio.com",
}

type FetchMarketplaceAvailableAddOnExtensionPathParams struct {
    AvailableAddOnSid string `pathParam:"style=simple,explode=false,name=AvailableAddOnSid"`
    Sid string `pathParam:"style=simple,explode=false,name=Sid"`
    
}

type FetchMarketplaceAvailableAddOnExtensionSecurity struct {
    AccountSidAuthToken shared.SchemeAccountSidAuthToken `security:"scheme,type=http,subtype=basic"`
    
}

type FetchMarketplaceAvailableAddOnExtensionRequest struct {
    ServerURL *string 
    PathParams FetchMarketplaceAvailableAddOnExtensionPathParams 
    Security FetchMarketplaceAvailableAddOnExtensionSecurity 
    
}

type FetchMarketplaceAvailableAddOnExtensionResponse struct {
    ContentType string 
    StatusCode int64 
    PreviewMarketplaceAvailableAddOnAvailableAddOnExtension *shared.PreviewMarketplaceAvailableAddOnAvailableAddOnExtension 
    
}

File: SDKs/twilio.com/twilio_preview/1.36.0/typescript/src/sdk/models/operations/fetchmarketplaceavailableaddonextension.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const FETCHMARKETPLACEAVAILABLEADDONEXTENSION_SERVERS = [
	"https://preview.twilio.com",
];



export class FetchMarketplaceAvailableAddOnExtensionPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=AvailableAddOnSid" })
  availableAddOnSid: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=Sid" })
  sid: string;
}


export class FetchMarketplaceAvailableAddOnExtensionSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class FetchMarketplaceAvailableAddOnExtensionRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: FetchMarketplaceAvailableAddOnExtensionPathParams;

  @Metadata()
  security: FetchMarketplaceAvailableAddOnExtensionSecurity;
}


export class FetchMarketplaceAvailableAddOnExtensionResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  previewMarketplaceAvailableAddOnAvailableAddOnExtension?: shared.PreviewMarketplaceAvailableAddOnAvailableAddOnExtension;
}
